### PR TITLE
wip conversion to libcbv2g

### PR DESCRIPTION
**Pull request by chardin-cpi**
Need to use a supported library under active development.

see https://github.com/ChargePoint/wireshark-v2g/issues/56

### CODE REVIEW

**Review by jhart-cpi — APPROVED** [review]

File: dissector/v2g.lua [hunk 1]
```diff
@@ -12,212 +12,245 @@ local V2GTP_HEADER_LENGTH = 8
...
-        end
-    end
+	local payload_type_name = get_sdp_payload_type(buffer(2, 2):uint())
+	local sdp_pay_len = buffer(4, 4):uint()
+
+	local subtree = tree:add(v2gtp_protocol, buffer(), "V2G Transfer Protocol")
+	subtree:add(SDP["version"], buffer(0, 1)):append_text(append_paren_text(get_sdp_version(buffer(0, 1):uint())))
+	subtree
+		:add(SDP["version_inverted"], buffer(1, 1))
+		:append_text(append_paren_text(get_sdp_version(buffer(0, 1):uint())))
```
Comment (barsnick) [inline dissector/v2g.lua:150]: As mentioned elsewhere: You cannot use the same check for both fields: The first byte **must** be the non-inverted version, the second one the inverted version.

Here, you check both bytes with the same check. So any of 0x0101, 0x01FE, 0xFE01, 0xFEFE would pass. Only the second of these is correct.

Furthermore: You add a subtree "version_inverted" for `buffer(1, 1)`, but pass `buffer(0, 1)` to the check. That's even more wrong, but probably a copy-paste error.

I would make `get_sdp_version()` check only for 0x01, and here pass the _inversion_ of `buffer(1, 1)` to that function, something like ``get_sdp_version(bit.bxor(buffer(1,1):uint(), 0xFF))`. (No reason to use `bit32` either, from my experiments.)

File: dissector/v2g.lua [hunk 2]
```diff
@@ -12,212 +12,245 @@ local V2GTP_HEADER_LENGTH = 8
...
-    response_secc_ip_addr = ProtoField.ipv6("sdp.ResponseSeccAddr", "SDP Response IP Address", base.HEX),
-    response_secc_port = ProtoField.uint16("sdp.ResponseSeccPort", "SDP Response Port", base.DEC),
-    response_security = ProtoField.uint8("sdp.ResponseSecurity", "SDP Response Security", base.HEX),
-    response_transport_proto = ProtoField.uint8("sdp.ResponseTransportProtocol", "SDP Response Transport Protocol", base.HEX),
-    response_emsp_ids = ProtoField.string("sdp.ResponseEMSPIDs", "EMSP IDs"),
+	-- header (8 bytes)
+	version = ProtoField.uint8("sdp.ProtocolVersion", "SDP Protocol Version", base.HEX),
+	version_inverted = ProtoField.uint8("sdp.InvertedProtocolVersion", "SDP Inverted Protocol Version", base.HEX),
+	payload_type = ProtoField.uint16("sdp.PayloadType", "SDP Payload Type", base.HEX),
+	payload_length = ProtoField.uint32("sdp.PayloadLength", "SDP Payload Length", base.HEX),
```
Comment (barsnick) [inline dissector/v2g.lua:19]: It feels unnatural to display payload length in HEX.
```suggestion
	payload_length = ProtoField.uint32("sdp.PayloadLength", "SDP Payload Length", base.DEC),
```

File: dissector/v2g.lua [hunk 3]
```diff
@@ -12,212 +12,245 @@ local V2GTP_HEADER_LENGTH = 8
...
+		[0x9005] = "SDP EMSP RESPONSE",
+	}
+
+	if set_contains(types, type) then
+		return types[type]
+	end
+
+	if type < 0x8000 then
+		type_name = "RESERVED"
+	elseif type >= 0x8001 and type <= 0x8FFF then
```
Comment (barsnick) [inline dissector/v2g.lua:105]: By addition of the ISO -20 codes, this now starts from 0x8007, but you return early anyway.
```suggestion
	elseif type >= 0x8007 and type <= 0x8FFF then
```

**Review by barsnick — CHANGES_REQUESTED** [review]

**Comment by JeremyWhaling** [comment]
Just FYI, the base v2g.lua file appears to be prior to my pull request. In other words, the ESDP/J3400/ISO 15118-200 support isn't included. I can help add it back if you would like.

**Comment by JeremyWhaling** [comment]
Note once the dust settles, I also have an improvement to the "dissect_all_tlvs_in_packet" function: I added support to handle malformed TLVs without throwing a Lua error and instead flags the TLV as malformed within Wireshark.

**Pull request closed without merging**